### ISSUE TITLE
unordered_map::iterator::operator*() returns rvalue instead of reference

### ISSUE THREAD

**Issue by mrdimfox** [ETLCPP/etl#148]
If you want to iterate throw a `unordered_map` like this:
```cpp
for (auto& [k, v] : map) {
    // ...
}
```

you can't. Compilation fails because of the return type of this operator:

```cpp
std::pair<const TKey, T> operator *()
{
    return inode->key_value_pair;
}
```

It should be a reference, I guess.

Compiler CLang 8.0.1. Same for gcc arm noneabi. ETL 14.29.3.

**Comment by jwellbelove** [comment]
Thanks, good spot!

**Comment by jwellbelove** [comment]
Fixed 14.29.4
